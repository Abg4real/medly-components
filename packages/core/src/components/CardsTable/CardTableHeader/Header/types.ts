import { SortOrder } from '../../types';
import { WithThemeProp } from '@medly-components/utils/src';

export interface Props extends WithThemeProp{
    sortField?: string;
    field?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
    enableSorting?: boolean;
    title: string;
}    