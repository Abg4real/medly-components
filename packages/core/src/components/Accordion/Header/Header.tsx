import { ExpandMoreIcon } from '@medly-components/icons';
import { WithStyle } from '@medly-components/utils';
import React, { useCallback, useContext } from 'react';
import { AccordionContext } from '../AccordionContext';
import * as Styled from './Header.styled';

export const Header: React.SFC & WithStyle = React.memo(({ children, ...restProps }) => {
    const [isActive, setActiveState] = useContext(AccordionContext),
        handleClick = useCallback(() => setActiveState(val => !val), []);

    return (
        <Styled.Wrapper aria-expanded={isActive} isActive={isActive} onClick={handleClick} {...restProps}>
            {children}
            <ExpandMoreIcon size="M" />
        </Styled.Wrapper>
    );
});

Header.displayName = 'Header';
Header.Style = Styled.Wrapper;
