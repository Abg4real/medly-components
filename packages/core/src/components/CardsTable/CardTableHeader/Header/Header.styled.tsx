import { SvgIcon } from '@medly-components/icons';
import { css, styled } from '@medly-components/utils';
import Card from '../../../Card';
import Text from '../../../Text';
import { Props } from './types';


const getTextStyle = ({ theme }: Props) => {
    const {variants, defaults, weights} = theme.font

    return css`
        color: ${defaults.color};
        text-transform: uppercase;
        font-size: ${variants.h5.fontSize};
        letter-spacing: ${variants.h5.letterSpacing};
        font-weight: ${weights[variants.h5.fontWeight]};
        display: flex;
        align-items: center;
    `;
};

export const HeaderStyled = styled(Card)<Props>`
    padding: 0 10px;
    ${Text.Style} {
        ${getTextStyle};
    }

    ${SvgIcon} {
        cursor: pointer;
        margin-right: 1.0rem;
        padding-left: 1.0rem;
    }
`;
