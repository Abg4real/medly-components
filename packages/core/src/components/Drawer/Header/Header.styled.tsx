import { rgba } from 'polished';
import { styled } from '@medly-components/utils';
import { CloseIcon } from '@medly-components/icons';

export const HeaderStyled = styled('div')<{ showShadow: boolean }>`
    padding: 1.6rem 2.4rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    box-shadow: ${({ theme, showShadow }) => showShadow && `0 1.8rem 1.6rem -1.6rem ${rgba(theme.colors.grey[400], 0.2)}`};
`;

export const CloseIconStyled = styled(CloseIcon)`
    border: 1px solid ${({ theme }) => theme.colors.grey[300]};
`;
