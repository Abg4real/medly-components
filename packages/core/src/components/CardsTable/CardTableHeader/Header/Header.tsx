import { ArrowDropDownIcon, ArrowDropUpIcon, DropdownIcon } from "@medly-components/icons/src";
import { WithStyle } from "@medly-components/utils/src";
import React, { SFC, useCallback, useEffect, useMemo, useRef, useState } from "react";
import Card from "../../../Card";
import Text from "../../../Text";
import { HeaderStyled } from "./Header.styled";
import { Props } from "./types";


export const Header: SFC<Props> & WithStyle = React.memo( props => {

    const { sortField, field, onSortChange, enableSorting, title } = props;

    const [sortState, setSortState] = useState<'none' | 'asc' | 'desc'>('none');

    const cellEl = useRef(null);

    useEffect(() => {
        if (sortField !== field) setSortState('none');
    }, [sortField]);
    
    const handleSortIconClick = useCallback(() => {
        const order = sortState === 'asc' ? 'desc' : 'asc';
        setSortState(order);
        onSortChange(field, order);
    }, [sortState, onSortChange, field]);

    const sortIcon = useMemo(() =>
            sortField !== field ? (
                <DropdownIcon size="XS" onClick={handleSortIconClick} />
            ) : sortState === 'desc' ? (
                <ArrowDropDownIcon size="M" onClick={handleSortIconClick} />
            ) : (
                <ArrowDropUpIcon size="M" onClick={handleSortIconClick} />
            ),
        [handleSortIconClick, sortField, field, sortState]
    );
    
    return (
        <HeaderStyled 
            ref={cellEl} 
            flowDirection="horizontal" 
            variant="flat"
            fullWidth 
            withPadding={false}
            title={title}
        >
            <Card key={field} variant="flat" alignItems="left">
                <Text>
                    {title}
                    {enableSorting && sortIcon}
                </Text>
            </Card>

        </HeaderStyled>
    )
 


});
Header.displayName = 'Header';
Header.Style = HeaderStyled;

