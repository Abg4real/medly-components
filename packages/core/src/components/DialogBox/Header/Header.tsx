import { isValidStringOrNumber, WithStyle } from '@medly-components/utils';
import React, { useContext, useEffect, useRef } from 'react';
import Text from '../../Text';
import { DialogBoxContext } from '../DialogBox.context';
import * as Styled from './Header.styled';

export const Header: React.FC & WithStyle = React.memo(({ children }) => {
    const { id, setHeaderHeight, isSmallScreen, scrollState } = useContext(DialogBoxContext);
    const headerRef = useRef<HTMLDivElement>();

    useEffect(() => {
        headerRef.current && setHeaderHeight(headerRef.current.offsetHeight);
    }, [headerRef.current, isSmallScreen]);

    return (
        <Styled.Header ref={headerRef} {...{ scrollState, id: `${id}-header` }}>
            {React.Children.map(children, c => {
                return isValidStringOrNumber(c) ? <Text textVariant={isSmallScreen ? 'h3' : 'h2'}>{c}</Text> : c;
            })}
        </Styled.Header>
    );
});
Header.displayName = 'Header';
Header.Style = Styled.Header;
