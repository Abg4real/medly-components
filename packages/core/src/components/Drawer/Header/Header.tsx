import { isValidStringOrNumber, WithStyle } from '@medly-components/utils';
import React, { useContext } from 'react';
import Text from '../../Text';
import { DrawerContext } from '../Drawer.context';
import { CloseIconStyled, HeaderStyled } from './Header.styled';

export const Header: React.FC & WithStyle = React.memo(({ children, ...restProps }) => {
    const { id, scrollState, onClose } = useContext(DrawerContext);
    return (
        <HeaderStyled showShadow={!scrollState.scrolledToTop} {...restProps} id={`${id}-header`}>
            {React.Children.map(children, c => {
                return isValidStringOrNumber(c) ? <Text textVariant="h3">{c}</Text> : c;
            })}
            <CloseIconStyled onClick={onClose} size="XS" variant="solid" id={`${id}-close-icon`} />
        </HeaderStyled>
    );
});

Header.displayName = 'DrawerHeader';
Header.Style = HeaderStyled;
