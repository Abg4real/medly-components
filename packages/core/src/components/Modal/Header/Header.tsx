import { isValidStringOrNumber, WithStyle } from '@medly-components/utils';
import React from 'react';
import Text from '../../Text';
import * as Styled from './Header.styled';
import { Props } from './types';

export const Header: React.SFC<Props> & WithStyle = React.memo(props => (
    <Styled.Header {...{ ...props, id: `${props.id}-header` }}>
        {React.Children.map(props.children, c => {
            return isValidStringOrNumber(c) ? <Text textVariant="h2">{c}</Text> : c;
        })}
    </Styled.Header>
));
Header.displayName = 'Header';
Header.Style = Styled.Header;
