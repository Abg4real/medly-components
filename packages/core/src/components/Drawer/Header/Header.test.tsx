import { render } from '@test-utils';
import React from 'react';
import Header from './';

describe('Drawer Header Test', () => {
    it('should render text if string is passed', () => {
        const { container } = render(<Header>Demo Header</Header>);
        expect(container).toMatchSnapshot();
    });

    it('should render component if its passed', () => {
        const { container } = render(
            <Header>
                <div>Render DIV</div>
            </Header>
        );
        expect(container).toMatchSnapshot();
    });
});
