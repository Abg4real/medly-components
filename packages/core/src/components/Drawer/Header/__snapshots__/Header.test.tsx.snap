// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Header Test should render component if its passed 1`] = `
.c1 {
  overflow: visible;
  font-size: 1.6rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
  padding: 0.8rem;
  border-radius: 50%;
  background-color: #ffffff;
}

.c1 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #13181D;
}

.c0 {
  padding: 1.6rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  border: 1px solid #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <div>
      Render DIV
    </div>
    <svg
      bgColor="#ffffff"
      class="c1 c2"
      fill="none"
      height="1em"
      iconColor="#13181D"
      variant="solid"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
        fill="#000"
        fill-opacity=".54"
      />
    </svg>
  </div>
</div>
`;

exports[`Drawer Header Test should render text if string is passed 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 400;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  line-height: 3.2rem;
  text-align: initial;
}

.c2 {
  overflow: visible;
  font-size: 1.6rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
  padding: 0.8rem;
  border-radius: 50%;
  background-color: #ffffff;
}

.c2 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #13181D;
}

.c0 {
  padding: 1.6rem 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border: 1px solid #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      Demo Header
    </h3>
    <svg
      bgColor="#ffffff"
      class="c2 c3"
      fill="none"
      height="1em"
      iconColor="#13181D"
      variant="solid"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
        fill="#000"
        fill-opacity=".54"
      />
    </svg>
  </div>
</div>
`;
